config/app/api/openapi/registry.yml

```
@@ -0,0 +1,117 @@
+                400:
+                    description: 'Bad request.'
+                    content:
+                        application/json:
+                            schema:
+                                $ref: '#/components/schemas/ApiErrorMessages'
+                default:
+                    description: 'Expected response to a bad request.'
+                    content:
+                        text/plain:
+                            schema:
+                                type: string
+        put: *paths/private/application-config/post # $ref is not (officially) supported for operation objects
+
+    '/private/application/disconnect':
+        post:
+            summary: 'Disconnects Application from a Tenants store.
+                      Finds Application configuration and removes it.'
+            parameters:
+                -   name: X-Store-Reference
+                    in: header
+                    required: true
+                    description: 'Reference of the Tenants store.'
+                    schema:
+                        type: string
+            responses:
+                204:
+                    description: 'App is disconnected.'
+                400:
+                    description: 'Bad request.'
+                    content:
+                        application/json:
+                            schema:
+                                $ref: '#/components/schemas/ApiErrorMessages'
+                default:
+                    description: 'Expected response to a bad request.'
+                    content:
+                        application/json:
+                            schema:
+                                $ref: '#/components/schemas/ApiErrorMessages'
+components:
+    schemas:
+        ConfigurationApiResponse:
+            type: object
+            additionalProperties: true
+        ConfigurationApiRequest:
+            properties:
+                data:
+                    type: object
+                    required:
+                        - attributes
+                    properties:
+                        attributes:
+                            type: object
+                            required:
+                                - configuration
+                            properties:
+                                configuration:
+                                    description: 'Contains JSON encoded configuration.'
+                                    type: string
+        ApiErrorMessage:
+            properties:
+                code:
+                    type: string
+                detail:
+                    type: string
+                status:
+                    type: string
+        ApiErrorMessages:
+            properties:
+                data:
+                    type: array
+                    items:
+                        $ref: '#/components/schemas/ApiErrorMessage'
+    parameters:
```

```suggestion
    headers:
```
OpenAPI spec allows for `components/headers`: https://spec.openapis.org/oas/v3.0.3#components-object

---

this isn't possible unfortunately: headers object can't be used as a reference for parameters object - getting invalid schema this way

---

You're right that using `headers` here will make the schema invalid. Let's keep this as-is.